source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateH0443.ts

```diff
@@ -0,0 +1,172 @@
+      const contactOwner = owners.find(x => x.isPrimaryContact === true);
+
+      const h0443Data: H0443Data = {
+        file_name: file.fileName || '',
+        file_number: file.fileNumber || '',
+        project_number: file.project?.code || '',
+        project_name: file.project?.description || '',
+        properties: properties !== undefined ? properties.map<Property>(x => getProperty(x)) : [],
+        owner_names: owners.map<string>(x => getOwnerName(x)) || [],
+        owner_contact: contactOwner !== undefined ? getOwnerContact(contactOwner) : null,
+        property_coordinator: {
+          first_name: coordinatorPerson?.firstName || '',
+          last_name: coordinatorPerson?.surname || '',
+        },
+        property_agent: {
+          first_name: agentPerson?.firstName || '',
+          last_name: agentPerson?.surname || '',
+          phone:
+            agentPerson?.contactMethods?.find(
+              x => x.contactMethodType?.id === ContactMethodTypes.WorkPhone,
+            )?.value || '',
+        },
+      };
+
+      const generatedFile = await generate({
+        templateType: FormDocumentType.H0443,
+        templateData: h0443Data,
+      });
+      generatedFile?.status === ExternalResultStatus.Success!! &&
+        generatedFile?.payload &&
+        showFile(generatedFile?.payload);
+    }
+  };
+  return generateLetter;
+};
+
+function getOwnerName(owner: Api_AcquisitionFileOwner): string {
+  if (owner.isOrganization) {
+    var corpName: string = owner.lastNameAndCorpName || '';
+    if (owner.incorporationNumber) {
+      corpName += ` (Inc. No. ${owner.incorporationNumber})`;
+    } else {
+      corpName += ` (Reg. No. ${owner.registrationNumber})`;
+    }
+    return corpName;
+  } else {
+    var personName: string = `${owner.givenName} ${owner.lastNameAndCorpName}`;
+    return personName;
+  }
+}
+
+function getOwnerContact(owner: Api_AcquisitionFileOwner): PersonWithPhone {
+  if (owner.isOrganization) {
+    return {
+      last_name: owner.lastNameAndCorpName || '',
+      first_name: owner.otherName || '',
+      phone: owner.contactPhoneNum || '',
+    };
+  } else {
+    return {
+      last_name: owner.lastNameAndCorpName || '',
+      first_name: owner.givenName || '',
+      phone: owner.contactPhoneNum || '',
+    };
+  }
+}
+
+function getPropertyAddress(address: Api_Address): Address {
+  return {
+    municipality: address.municipality || '',
+    postal_code: address.postal || '',
+    province: address.province?.description || '',
+    street_address_1: address.streetAddress1 || '',
+    street_address_2: address.streetAddress2 || '',
+    street_address_3: address.streetAddress3 || '',
+  };
+}
+
+function getProperty(property: Api_Property): Property {
+  return {
+    address: property.address !== undefined ? getPropertyAddress(property.address) : null,
+    legal_description: property.landLegalDescription || '',
+    pid: property.pid?.toString() || '',
+  };
+}
+
+interface Person {
```

Comment:
nit: co-locating this with the hook that uses this makes sense, but maybe not in the same file.